### PR TITLE
pin-2680: New version lifecycle

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala:59]: This double step is not a very clean solution, but right now it is not possible to directly create a waiting for approval version because the business logic and the checks are performed on the management

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala:59]: Until we have the logic inside management, I don't see a different solution from what you have implemented.
Just a minor suggestion: I would pass the version explicitly in functions `changeToWaitForApproval` and `createWaitForApproval` like we do in `activate`.

**Review by beetlecrunch — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

**Pull request merged**
